src/spec.c

```diff
@@ -2476,10 +2505,7 @@ void IndexSpec_DropLegacyIndexFromKeySpace(IndexSpec *sp) {
       Redis_DeleteKey(ctx.redisCtx, IndexSpec_GetFormattedKey(ctx.spec, fs, INDEXFLD_T_GEO));
     }
   }
-  RedisModuleString *str =
-      RedisModule_CreateStringPrintf(ctx.redisCtx, INDEX_SPEC_KEY_FMT, ctx.spec->name);
-  Redis_DeleteKey(ctx.redisCtx, str);
-  RedisModule_FreeString(ctx.redisCtx, str);
+  HiddenString_DropFromKeySpace(ctx.redisCtx, INDEX_SPEC_KEY_FMT, sp->specName);
```

Comment:
I would avoid this trait for hidden string
IMO it should not be aware of key space, RDB etc..